_posts/2020-03-04-eda-architecture2.md

```diff
@@ -0,0 +1,73 @@
+---
+layout: post
+title:  "An event driven architecture — part 2"
+excerpt: "Core design principles."
+date:   2020-03-04 12:53:00 +0100
+categories: architecture
+tags:  services event bus architecture
+---
+
+In the previous post in this series, we explained why we ditched our old architecture based on synchronous REST services for a completely asynchronous event driven architecture.
+
+Today, we address the core design principles that were crucial in the success of this enterprise.
+
+## Business Services and Data Processing Services
+
+We make a distinction between *Business Services* and *Data Processing Services* (a.k.a. utility services) to cleanly separate business logic from data processing complexity.
+
+DP Services are expected to be  **pure, stateless, services** that provide some kind of algorithmic data processing (computations, transformations, …). Moreover, they are also context free: they should not depend on business rules, assumptions or external data sources. All they need to do their processing must be in the message they receive. They should not have to query a tier to get more data. DP services are kind of *universal* libraries and can even be provided by tiers.
```

Comment:
Could you illustrate Business Services and Data Processing Services with examples from our architecture, as well as maybe examples from a easily understandable point of reference? (such as an e-commerce example, that would echo the event `ShoppingCartValidated` below)